o1vm/src/pickles/prover.rs

```diff
@@ -280,6 +288,29 @@ where
     fr_sponge.absorb(&zeta_omega_evaluations.error);
     // FIXME: add selectors
 
+    // Compute ft(X) = \
+    //   (1 - ζ^n) \
+    //    (t_0(X) + ζ^n t_1(X) + ... + ζ^{kn} t_{k}(X))
+    // where Σ_i t_i(X) X^{i n} = t(X), and t(X) is the quotient polynomial.
+    // At the end, we get the (partial) evaluation of the constraint polynomial
+    // in ζ.
+    let ft: DensePolynomial<G::ScalarField> = {
+        let evaluation_point_to_domain_size = zeta.pow([domain.d1.size]);
+        // Compute Σ_i t_i(X) ζ^{i n}
+        // First we split t in t_i, and we reduce to degree (n - 1) after using `linearize`
+        let t_chunked: DensePolynomial<G::ScalarField> = quotient_poly
+            .to_chunked_polynomial(DEGREE_QUOTIENT_POLYNOMIAL as usize, domain.d1.size as usize)
+            .linearize(evaluation_point_to_domain_size);
+        // -Z_H = (1 - ζ^n)
+        let minus_vanishing_poly_at_zeta = -domain.d1.vanishing_polynomial().evaluate(&zeta);
+        // Multiply the polynomial Σ_i t_i(X) ζ^{i n} by -Z_H(ζ)
+        // (the evaluation in ζ of the vanishing polynomial)
+        t_chunked.scale(minus_vanishing_poly_at_zeta)
+    };
```

Comment:
I am confused, why are we evaluating this ?
We should evaluate the chunks of t, not scale by the vanishsing poly ?
Also, if we do chunking, at this point we should have access to the chunks (since we commited to them), so we should not call to chunk one more time
And if we want to evaluate the constraint poly, we should not go through t, we have a direct access to it